### PR TITLE
Added optimizers for increasing batch size to 32 and fixed a minor bug in loss mask

### PR DESCRIPTION
**Pull request by RishuG-work**
### What does this PR do?

Added optimizers for increasing batch size to 32 and fixed a minor bug in loss mask

### CODE REVIEW

File: orby/data/convert_screenspot_sft.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
+# Copyright 2024 Bytedance Ltd. and/or its affiliates
```
Comment (ligang-orby) [inline orby/data/convert_screenspot_sft.py:1]: Seems some files in this PR were already submitted in previous PR. You can merge the changes from main and not submit these files again.

git checkout main
git pull
git checkout fix-vlm-sft
git merge main

There might be some conflicts to solve

File: orby/trainer/fsdp_sft_trainer.py [hunk 2]
```diff
@@ -335,6 +367,37 @@ def _compute_loss_and_backward(self, batch, do_backward=True):
         raw_prompt_ids = batch["raw_prompt_ids"]
         multi_modal_inputs = batch.get("multi_modal_inputs", {})
         
+        # MINIMAL DEBUG: Log sequence lengths and memory
```
Comment (ligang-orby) [inline orby/trainer/fsdp_sft_trainer.py:370]: Remove these debug codes. Keep a local version if needed.

File: orby/trainer/fsdp_sft_trainer.py [hunk 3]
```diff
@@ -488,6 +550,14 @@ def _compute_loss_and_backward(self, batch, do_backward=True):
     def training_step(self, batch: TensorDict):
         self.fsdp_model.train()
 
+
+        # MINIMAL DEBUG: Check memory before processing
```
Comment (ligang-orby) [inline orby/trainer/fsdp_sft_trainer.py:554]: remove

File: orby/trainer/fsdp_sft_trainer.py [hunk 4]
```diff
@@ -680,20 +769,28 @@ def main(config):
     # build tokenizer and datasets first
     local_model_path = copy_to_local(src=config.model.partial_pretrain, verbose=True)
     tokenizer = hf_tokenizer(local_model_path, trust_remote_code=config.model.trust_remote_code)
-    processor = hf_processor(local_model_path, **config.data.get("processor", {}))
+    processor = hf_processor(local_model_path, **config.get("processor", {}))
+    #print(f"Processor type: {type(processor)}")
```
Comment (ligang-orby) [inline orby/trainer/fsdp_sft_trainer.py:773]: remove debug codes

File: orby/trainer/fsdp_sft_trainer.py [hunk 5]
```diff
@@ -630,7 +703,23 @@ def fit(self):
                 metric = self.training_step(data)
                 if rank == 0:
                     tracking.log(data=metric, step=global_step)
+                # Validation every 25 steps
```
Comment (ligang-orby) [inline orby/trainer/fsdp_sft_trainer.py:706]: you can try to add save_freq (checkpoint save freq) and test_freq (eval freq) like ppo trainer: https://github.com/orby-ai-engineering/verl/blob/main/orby/scripts/run_qwen2_5_vl_7b_grpo.sh#L55

**Review by ligang-orby — APPROVED** [review]
